src/Pact/Analyze/Types/Shared.hs

```diff
@@ -1002,7 +1002,7 @@ instance
     Just (a, Object as) -> Just $ Object $ SCons SSymbol (Column sing a) as
 
 withSymVal :: SingTy a -> (SymVal (Concrete a) => b) -> b
-withSymVal = withDict . singMkSymVal
+withSymVal a = withDict $ singMkSymVal a
   where
 
     singMkSymVal :: SingTy a -> Dict (SymVal (Concrete a))
```

Comment:
why were these above changed to have explicit application? If it's preferential, I Would consider this a step back.

Reply:
The constraint in the type is impredicative. Adding the argument causes the function to `withDict` being fully applied. This changes the order how types are resolved during type checking. Future version are more picky about impredicativity.

https://gitlab.haskell.org/ghc/ghc/-/wikis/migration/8.10#ghc-is-pickier-about-impredicative-uses-of-constraints